tests/sentry/api/endpoints/test_project_stacktrace_link.py

```diff
@@ -0,0 +1,104 @@
+        self.project = self.create_project(
+            name="foo", organization=self.org, teams=[self.create_team(organization=self.org)]
+        )
+
+        self.integration = Integration.objects.create(provider="example", name="Example")
+        self.integration.add_organization(self.org, self.user)
+        self.oi = OrganizationIntegration.objects.get(integration_id=self.integration.id)
+
+        self.repo = self.create_repo(project=self.project, name="getsentry/sentry",)
+        self.repo.integration_id = self.integration.id
+        self.repo.provider = "example"
+        self.repo.save()
+
+        self.config = RepositoryProjectPathConfig.objects.create(
+            organization_integration=self.oi,
+            project=self.project,
+            repository=self.repo,
+            stack_root="/usr/src/getsentry",
+            source_root="",
+        )
+        self.filepath = "/usr/src/getsentry/src/sentry/src/sentry/utils/safe.py"
+        self.url = reverse(
+            "sentry-api-0-project-stacktrace-link",
+            kwargs={
+                "organization_slug": self.project.organization.slug,
+                "project_slug": self.project.slug,
+            },
+        )
+
+    def test_no_filepath(self):
+        self.login_as(user=self.user)
+
+        response = self.client.get(self.url, format="json")
+        assert response.status_code == 400, response.content
+
+    def test_no_configs(self):
+        self.login_as(user=self.user)
+        # new project that has no configurations set up for it
+        project = self.create_project(
+            name="bloop", organization=self.org, teams=[self.create_team(organization=self.org)],
+        )
+
+        path = reverse(
+            "sentry-api-0-project-stacktrace-link",
+            kwargs={"organization_slug": project.organization.slug, "project_slug": project.slug},
+        )
+        url = u"{}?file={}".format(path, self.filepath)
+
+        response = self.client.get(url, format="json")
+        assert response.status_code == 200, response.content
+        assert not response.data["config"]
+
+    def test_config_but_no_source_url(self):
+        self.login_as(user=self.user)
+        url = u"{}?file={}".format(self.url, self.filepath)
+
+        response = self.client.get(url)
+
+        assert response.status_code == 200, response.content
+        assert response.data["config"] == {
+            "id": six.text_type(self.config.id),
+            "repo": self.repo.name,
+            "project": self.project.slug,
+            "sourceRoot": self.config.source_root,
+            "stackRoot": self.config.stack_root,
+            "organizationIntegration": six.text_type(self.oi.id),
+            "defaultBranch": None,
+        }
+        assert not response.data["source_url"]
+
+    def test_config_and_source_url(self):
+        self.login_as(user=self.user)
+        url = u"{}?file={}".format(self.url, self.filepath)
+
+        with mock.patch.object(
```

Comment:
not sure this is the best test here since it's pretty limited, but wanted to have it for now will waiting on https://github.com/getsentry/sentry/pull/21454